core/src/metadata_json.rs

```diff
@@ -188,7 +229,25 @@ async fn fetch_metadata_json(
         }
     })
     .retry(&ExponentialBuilder::default())
-    .await
+    .notify(|_e, _d| {
+        retries.fetch_add(1, Ordering::Relaxed);
+    })
+    .await;
+
+    let time_elapsed = start.elapsed().as_secs();
+
+    let retries = retries.load(Ordering::Relaxed);
```

Comment:
backon give you access to the number of attempts?

Reply:
The function is a bit hard to follow with the match then decorating with extra context. Maybe do timing in the caller function and then have error respond with retries if can be retrieved from backon. If not then we can skip writing that to the db the status code is more informative. We can already tweak retries using backon so we know its max and timeout.

Reply:
backon doesnt give access to these , so dropping this from db